### PR TITLE
add travis CI

### PR DESCRIPTION
**Pull request by jonasschnelli**
Got it running after 51 tries.
Took me a while to figure out how to support C++11 and Qt5.x.
Solution was to use Docker (for Ubuntu/Linux).

Native OSX builds are also working.

https://travis-ci.org/jonasschnelli/dbb-app/jobs/84048227

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+language: c
```
Comment (lclc) [inline .travis.yml:1]: cpp

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+sudo: required
```
Comment (lclc) [inline .travis.yml:2]: If sudo usage is detected it is automatically set to yes and not build in the container-based Travis infrastructure (which is faster): http://docs.travis-ci.com/user/workers/container-based-infrastructure/

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+services:
+    - docker
```
Comment (lclc) [inline .travis.yml:4]: You don't have a docker file (yet?).

File: .travis.yml (hunk 1 above)
Comment (jonasschnelli) [inline .travis.yml:1]: Huh... good point. Sorry for the (too) quick merge. Thanks for the review!

**Comment by lclc** [comment]
Yep, setting up or changing things in Travis can be quite a hassle.

I don't have experience using docker with Travis, but I can't see any docker calls as documented here: http://docs.travis-ci.com/user/docker/

**Comment by jonasschnelli** [comment]
I had to use Docker so it would use Ubuntu 14.04 (instead of 12.04). 12.04 has no Qt5 package. Installing Qt5 on Ubuntu 12.04 over ppa didn't work. Also the non-legacy platform (sudo = false) has Qt5 not as whitelisted package.

**Comment by lclc** [comment]
But I don't see any docker call? You just start the docker service, not booting a docker image?

For me it looks like you install the Qt5 package (http://www.ubuntuupdates.org/package/ubuntu_sdk_release/precise/main/base/libqt5core5) on Ubuntu 12.04 
here: https://github.com/digitalbitbox/dbb-app/pull/4/files#diff-354f30a63fb0907d4ad57269548329e3R16

(Also libqt5core5a -> libqt5core5)

Or maybe the Travis documentation regarding Docker is outdated.

**Comment by jonasschnelli** [comment]
Right. Not sure if i did it right. I would be happy if you could try it. I think if you do a PR with some .travis.yml changes it will build with that changed environment. Thanks in advance!

**Comment by lclc** [comment]
I should stay quite ;P

I'll see when I have time this weekend

**Pull request merged**